tinker/configuration.py

```diff
@@ -61,35 +73,89 @@ def iterate_generator(iterates: List[ConfigEntry]) -> Iterator[ConfigEntry]:
     """
-    return itertools.chain(*(config_generator(i) for i in iterates))
+    return itertools.chain(*(preprocess_config_generator(i) for i in iterates))
+
+
+def is_smqtk(value: Any) -> bool:
+    """Return `True` if `value` is a "smqtk" directive."""
+    return type(value) is dict and "smqtk" in value
 
 
-def config_generator(value: ConfigEntry) -> Iterator[ConfigEntry]:
+def smqtk_generator(smqtk_def: Dict[str, ConfigEntry]) -> Iterator[ConfigEntry]:
+    r"""
+    Find implementation of and instantiate a SMQTK class.
+
+    Instantiated with a config all values specified
+    by "smqtk" directive.
+    """
+
+    # get available implementations
+    # (from environment variable and entrypoint extension)
+    candidate_types = Pluggable.get_impls()
+
+    # check if our class matches any of the implementations
+    class_name = smqtk_def["class"]
+    matched_class = None  # TODO: can add typing here for matched_class
+    for candidate in candidate_types:
+        if candidate.__name__ == class_name:
+            matched_class = candidate
+            # TODO: add some sort of error checking if multiple classes
+            # have the same name, so we don't have any unexpected
+            # behavior
+            break
+    if matched_class is None:
+        raise ValueError("No SMQTK definition found: {}".format(class_name))
+
+    # explict check that matched_class is Configurable
+    if not issubclass(matched_class, Configurable):
+        raise ValueError("{} must be Configurable".format(class_name))
+
+    # Check with is_usable (inherits from Pluggable)
+    if not matched_class.is_usable():
+        raise ValueError("SMQTK impl {} not usable".format(class_name))
+
+    # TODO: maybe check to call is_valid_plugin
+
+    smqtk_config = smqtk_def.get("config", {})
+
+    smqtk_impl = matched_class.from_config(cast(Dict[str, Any], smqtk_config))
+    return smqtk_impl
+
+def process_config(value: ConfigEntry):
+    if is_smqtk(value):
+        return smqtk_generator(cast(SMQTKDirective, value)["smqtk"])
+    else:
+        return value
+
+def preprocess_config_generator(value: ConfigEntry) -> Iterator[ConfigEntry]:
     """
     Generate values for any type that may appear in the configuration object.
 
     This is the top-level "entry point" for meta-configuration expansion. It
-    dispatches on the type passed to it, invoking the proper processing routine
+    dispatches on the type passed to it, invoking the proper preprocessing routine
     to generate partial results. The function is ultimately recursive, as those
     processing routines (listed above) examine the contents of the values passed
     to them, invoking this function again as needed.
     """
     if is_iterate(value):
         return iterate_generator(cast(IterateDirective, value)["iterate"])
+    #elif is_smqtk(value):
+    #if is_smqtk(value):
+    #    return smqtk_generator(cast(SMQTKDirective, value)["smqtk"])
```

Comment:
If these lines need to be struck, please do so. If you need them retained as commented-out code, please add a header comment explaining why.

Reply:
Fixed in https://github.com/tinker-engine/tinker-engine/commit/cae201db22856facd034c1352dde236d39e17fd7